src/AzureClient/EntryPoint/EntryPoint.cs

```diff
@@ -115,9 +115,62 @@ private ArgumentValue GetArgumentValue(System.Reflection.ParameterInfo parameter
             {
                 return new ArgumentValue.Result(Newtonsoft.Json.JsonConvert.DeserializeObject<Result>(parameterValue)!);
             }
+            else if (parameterType.IsQArray())
+            {
+                var arrayType = parameterType.GenericTypeArguments.FirstOrDefault() ??
+                    throw new ArgumentException($"Could not get the type of array.");
+                if (arrayType == typeof(bool)) {
+                    var values = Newtonsoft.Json.JsonConvert.DeserializeObject<bool[]>(parameterValue)
+                        .Select(v => new ArgumentValue.Bool(v))
+                        .ToImmutableArray<ArgumentValue>();
+                    return ArgumentValue.Array.TryCreate(values, ArgumentType.Bool) ??
+                        throw new ArgumentException($"Could not create array of Bool for {parameter.Name}.");
+                }
+                else if (arrayType == typeof(double))
+                {
+                    var values = Newtonsoft.Json.JsonConvert.DeserializeObject<double[]>(parameterValue)
+                        .Select(v => new ArgumentValue.Double(v))
+                        .ToImmutableArray<ArgumentValue>();
+                    return ArgumentValue.Array.TryCreate(values, ArgumentType.Double) ??
+                        throw new ArgumentException($"Could not create array of Bool for {parameter.Name}.");
```

Comment:
This seems mostly duplicated with the version for `bool`, is it possible to extract into an extension method or similar?

Reply:
It is very similar, but in this case the string is deserialized into an array of `bool` which would cause the extension method to not be exactly the same.